src/dbnode/storage/series/buffer.go

```diff
@@ -551,20 +737,44 @@ func (b *dbBufferBucket) canRead() bool {
 }
 
 func (b *dbBufferBucket) needsReset(
+	now time.Time,
 	targetStart time.Time,
 ) bool {
+	if !b.isRealTime {
+		return false
+	}
+
 	return !b.start.Equal(targetStart)
 }
 
+func (b *dbBufferBucket) isStale(now time.Time) bool {
+	if b.isRealTime {
+		return false
+	}
+
+	return b.numWrites() > 0 && now.Sub(b.lastWrite()) > b.opts.RetentionOptions().FlushAfterNoMetricPeriod()
```

Comment:
What I call "stale" here is to prevent buckets from lying around for too long. Buckets that are stale haven't been written to for a while, so should be reclaimed, but if a bucket has been written to recently, it's likely that it will be written to again. It would be a waste to reclaim the bucket just to reinitialize one again on the next write.